eth/eventhandler/handlers.go

```diff
@@ -301,18 +315,21 @@ func validatorAddedEventToShare(
 		shareSecret = &bls.SecretKey{}
 		decryptedSharePrivateKey, err := rsaencryption.DecodeKey(operatorPrivateKey, encryptedKeys[i])
 		if err != nil {
-			return nil, nil, &MalformedEventError{
-				Err: fmt.Errorf("could not decrypt share private key: %w", err),
+			return &validatorShare, nil, &MalformedEventError{
+				Err:                     fmt.Errorf("could not decrypt share private key: %w", err),
+				IsInvalidEncryptedShare: true,
 			}
 		}
 		if err = shareSecret.SetHexString(string(decryptedSharePrivateKey)); err != nil {
-			return nil, nil, &MalformedEventError{
-				Err: fmt.Errorf("could not set decrypted share private key: %w", err),
+			return &validatorShare, nil, &MalformedEventError{
+				Err:                     fmt.Errorf("could not set decrypted share private key: %w", err),
+				IsInvalidEncryptedShare: true,
 			}
 		}
 		if !bytes.Equal(shareSecret.GetPublicKey().Serialize(), validatorShare.SharePubKey) {
-			return nil, nil, &MalformedEventError{
-				Err: errors.New("share private key does not match public key"),
+			return &validatorShare, nil, &MalformedEventError{
+				Err:                     fmt.Errorf("share private key does not match public key"),
+				IsInvalidEncryptedShare: true,
 			}
```

Comment:
can you please try to instead of returning malformed error, instead to just set the Invalid flag on the share, and later just ask if the share is invalid whereever you currently ask if there was a malformedError